packages/patternfly-react/src/components/Label/CompoundLabel.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Tooltip } from '../Tooltip';
+import { OverlayTrigger } from '../OverlayTrigger';
+import LabelWithTooltip from './LabelWithTooltip';
+
+class CompoundLabel extends React.Component {
+  generateTag = value => (
+    <LabelWithTooltip
+      key={value.id}
+      category={this.props.category}
+      value={value}
+      onDeleteClick={this.props.onDeleteClick}
+      truncate={this.props.valueTruncate}
+    />
+  );
+
+  render() {
+    const values = [...this.props.values];
+    if (values.length === 0) return <div />;
+    const categoryTooltip = (
+      <Tooltip id="tooltip">{this.props.category.label}</Tooltip>
+    );
+    return (
+      <ul className="category list-inline">
+        <OverlayTrigger placement="bottom" overlay={categoryTooltip}>
+          <div className="category-label">
+            {this.props.categoryTruncate(this.props.category.label)}
```

Comment:
Do we need to use the truncate function? Could we use a max-width and text-overflow instead?

Reply:
I think having truncate function is more flexible and if tooltip should be shown only when text is truncated as @mcarrano suggests, I think it will be easier to implement if I stick to truncate function.